File: packages/csharp/src/Errors/ApiException.cs
using System;

namespace Tesote.Sdk.Errors;

/// <summary>
/// Server returned a usable HTTP response with an error envelope.
/// Concrete subclasses correspond 1:1 to API <c>error_code</c> values.
/// </summary>
public class ApiException : TesoteException
{
    /// <summary>Construct with the full required-field set.</summary>
    public ApiException(
        string? message,
        string? errorCode,
        int httpStatus,
        string? requestId,
        string? errorId,
        int? retryAfter,
        string? responseBody,
        RequestSummary? requestSummary,
        int attempts,
        Exception? cause)
        : base(message, errorCode, httpStatus, requestId, errorId,
               retryAfter, responseBody, requestSummary, attempts, cause)
    {
    }
}

File: packages/java/src/main/java/com/tesote/sdk/errors/ApiException.java
package com.tesote.sdk.errors;

/**
 * Server returned a usable HTTP response with an error envelope. Concrete
 * subclasses correspond 1:1 to API {@code error_code} values.
 */
public class ApiException extends TesoteException {
    public ApiException(
            String message,
            String errorCode,
            int httpStatus,
            String requestId,
            String errorId,
            Integer retryAfter,
            String responseBody,
            RequestSummary requestSummary,
            int attempts,
            Throwable cause
    ) {
        super(message, errorCode, httpStatus, requestId, errorId,
                retryAfter, responseBody, requestSummary, attempts, cause);
    }
}
